browser-extension/mv3/src/service-worker/services/rulesManager.ts

```diff
@@ -89,6 +92,8 @@ const addExtensionRules = async (): Promise<void> => {
           );
         }
 
+        extensionRule.condition.excludedInitiatorDomains.push(...blockedDomains);
```

Comment:
Missed this in `RuleMatcher` too. Extension icon will turn green even though blocked